scripts/github_api

```diff
@@ -0,0 +1,118 @@
+#!/bin/python2
+from functools import partial
+import argparse
+import json
+import re
+import requests
+import sys
+
+CLIENT_ERROR = 422
+API_BASE_URL = "https://api.github.com/"
+
+
+def request(*args, **kwargs):
+    raise Exception("The '_setup_auth' function must be called once "
+                    "before a request can be made")
+
+
+def _setup_auth(user, password):
+    auth = (user, password)
+    global request
+    request = partial(_request, auth)
+
+
+def _request(auth, method, endpoint, **kwargs):
+    methods = {
+        'POST': requests.post
+    }
+
+    r = methods[method](endpoint, auth=auth, **kwargs)
+    response = json.loads(r.text)
+
+    try:
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as exc:
+        if r.status_code in [CLIENT_ERROR]:
+            error_msg = response['errors'][0]['message']
+            raise Exception(error_msg)
+        raise
+
+    return response
+
+
+class PullRequest:
+    """
+    https://developer.github.com/v3/pulls/
+    """
+    def __init__(self, repo):
+        self.repo = repo
+        self._endpoint = API_BASE_URL + "repos/%(repo)s/pulls"
+
+    def open(self, title, body, src_branch, dst_branch):
+        endpoint = self._endpoint % dict(repo=self.repo)
+        payload = {
+            "title": title,
+            "body": body,
+            "head": src_branch,
+            "base": dst_branch,
+        }
+
+        self._set_pr_info(request('POST', endpoint, data=json.dumps(payload)))
+
+    def _set_pr_info(self, pr_info):
+        for k, v in pr_info.items():
+            if type(v) != dict:
+                setattr(self, k, v)
```

Comment:
I am not sure what you are doing here. pr_info is the HTTP request response, right? (1) I'd prefer to mention "response" somewhere so that is clearer. (2) The response body from GitHub API is not always a JSON? In that case, you could use json()